templates/common/_base/files/configure-ovs-network.yaml

```diff
@@ -0,0 +1,134 @@
+    while [ $counter -lt 12 ]; do
+      iface=$(ip route | grep default | head -n 1 | grep -Eo "dev\s[^ ]+" | sed 's/dev\s//')
+      if [ ! -z "$iface" ]; then
+        echo "Default gateway interface found: ${iface}"
+        break
+      fi
+      counter=$((counter+1))
+      echo "No default route found on attempt: ${counter}"
+      sleep 5
+    done
+
+    if [ -z "$iface" ]; then
+      echo "Unable to find default gateway interface"
+      exit 1
+    fi
+
+    if [ "$iface" = "br-ex" ]; then
+      echo "Networking already configured and up for br-ex!"
+      exit 0
+    fi
+
+    # find the MAC of default interface to use for OVS internal port
+    # this prevents us from getting a different DHCP lease and dropping connection
+    iface_mac=$(cat /sys/class/net/${iface}/address)
+    if [ -z "$iface_mac" ]; then
+      echo "Unable to determine default interface MAC"
+      exit 1
+    fi
+
+    echo "MAC address found for iface: ${iface}: ${iface_mac}"
+
+    # find MTU on original iface
+    iface_mtu=$(ip link show ${iface} | grep -Eo "mtu\s[0-9]+" | sed 's/mtu\s//')
+    if [ -z "$iface_mtu" ]; then
+      echo "Unable to determine default interface MTU"
+      exit 1
+    fi
+
+    echo "MTU found for iface: ${iface}: ${iface_mtu}"
+
+    # store ipv4 address
+    iface_ip=""
+    if ip -4  addr show ${iface} |grep -Eo "([0-9]{1,3}[\.]){3}[0-9]{1,3}"; then
+      iface_ip=$(ip -4  addr show ${iface} | grep -Eo "([0-9]{1,3}[\.]){3}[0-9]{1,3}" | head -n 1)
+    fi
+
+    # create bridge
+    if ! nmcli connection show br-ex &> /dev/null; then
+      nmcli c add type ovs-bridge conn.interface br-ex con-name br-ex 802-3-ethernet.mtu ${iface_mtu} 802-3-ethernet.cloned-mac-address ${iface_mac}
+    fi
+
+    # find default port to add to bridge
+    if ! nmcli connection show ovs-port-phys0 &> /dev/null; then
+      nmcli c add type ovs-port conn.interface ${iface} master br-ex con-name ovs-port-phys0
+    fi
+
+    if ! nmcli connection show ovs-port-br-ex &> /dev/null; then
+      nmcli c add type ovs-port conn.interface br-ex master br-ex con-name ovs-port-br-ex
+    fi
+
+    if nmcli conn show --active | grep ${iface} | grep "Wired" ; then
+      old_conn=$(nmcli conn show --active | grep ${iface} | egrep '[0-9a-f]{8}-([0-9a-f]{4}-){3}[0-9a-f]{12}' -o)
+      nmcli conn down $old_conn
+    fi
+
+    if ! nmcli connection show ovs-if-phys0 &> /dev/null; then
+      nmcli c add type 802-3-ethernet conn.interface ${iface} master ovs-port-phys0 con-name ovs-if-phys0 \
+        connection.autoconnect-priority 100 802-3-ethernet.mtu ${iface_mtu}
+    fi
+
+    if ! nmcli connection show ovs-if-br-ex &> /dev/null; then
+      nmcli c add type ovs-interface slave-type ovs-port conn.interface br-ex master ovs-port-br-ex con-name \
+        ovs-if-br-ex 802-3-ethernet.mtu ${iface_mtu}
+    fi
+
+    # setting MTU on ovs iface does not seem to propagate to OVS
+    # https://gitlab.freedesktop.org/NetworkManager/NetworkManager/-/issues/206
```

Comment:
Since it should be fixed in 1.22.12, maybe we should request rhcos NM to be bumped (currently at 1.22.8 afaict).

Reply:
https://bugzilla.redhat.com/show_bug.cgi?id=1820052